### ISSUE TITLE
How to extract the right hex codes from a NAD amp?

### ISSUE THREAD

**Issue by Davincible** [tsprlng/nad-link#4]
Love this this program, works fantastic with most of the hex codes. However on my NAD 317 amp the on and off codes don't work, and I'd really like to have those working too.

How did you extract these codes? Or can I find them somewhere on the net..?

I figured maybe there was some info on the site you gave creds to, but that page doesn't exist anymore...

**Comment by tsprlng** [comment]
Ah yes, I see the page (which I did copy them from) has been taken down.

Luckily it's still available thanks to [the internet archive](http://web.archive.org/web/20130927174226/http://www.ping.uio.no/~mortehu/nadlink/).

I just used his [`irparse.c`](http://web.archive.org/web/20090611210417/http://www.ping.uio.no:80/~mortehu/nadlink/irparse.c) program on NAD's [Crestron code files](http://nadelectronics.com/software/#crestroncodes). Can't remember exactly how well it works, I think possibly the codes have to be reversed compared to how they come out of that tool (my script sends the bytes in left-to-right order). I might try again later and include all the codes myself in this repo.

Best of luck meanwhile! Thanks for writing, it's nice to know this has been slightly useful.

**Comment by tsprlng** [comment]
I've added a 'remote-codes' directory (334f690) which has all the converted output I could get. It wasn't that bad.

The thing was the `irparse` tool outputs correct binary but also some strange incorrect hex.

If you're curious, I did the extraction with the following awful script:
```bash
for f in *.ir; do
  ./irparse "$f" | perl -pe 's/([10]+)\[.+?\] ?/sprintf("%02x",oct("0b".$1))/ge' > $(basename "$f").hex
done
```

Cheers again and good luck :). Will hopefully tidy this up a bit more and duplicate the original explanation of the IR protocol later.
